common/src/main/java/com/wynntils/models/raid/type/RaidKind.java

```diff
@@ -55,6 +87,17 @@ public static RaidKind fromName(String name) {
         return null;
     }
 
+    public String majorIdFromBuff(String name) {
+        String[] split = name.split(" ");
+        if (split.length == 1) return null;
+
+        if (majorIdBuffs.containsKey(split[0])) {
+            return majorIdBuffs.get(split[0]).get(MathUtils.integerFromRoman(split[1]));
+        }
```

Comment:
This will error if a new major id is added to the game, but not the list (so basically it make older versions of the mod not work). I suggest you move the tier parsing and error handling to the model itself, and just do a simple lookup here.

Reply:
I don't see how it would error when a new major id is added to a buff or a new buff is added to the game? If the buff isn't present in that raid's list (e.g. Test II was added) then it will return null and the loop in the calling method will just continue to the next iteration as if it didn't find a major id from that buff at all. 

If a major id is added to an existing buff (e.g. Hollowed I begins giving the player Explosive Impact), then it will also return null and continue in the calling method's for loop as if the major id doesn't exist. I don't see where the error would be raised.

Reply:
What if "Hollowed 2's Guardian" buff is moved to "Hollowed 3"? Basically, you check the first get, but not the second one.

Reply:
Also, we want to keep the `type` classes as free from logic as possible; ideally they should just be simple Records, Enums, Interfaces etc; just keeping data but not any "code". That should be handled by the corresponding Model.

Reply:
Logic has been moved out of the Enum